src/src/com/microsoft/aad/adal/StorageHelper.java

```diff
@@ -264,28 +240,27 @@ public String decrypt(String value) throws GeneralSecurityException, IOException
 
         // get key version used for this data. If user upgraded to different
```

Comment:
Please refrain from throwing illegalArgumentExceptions when encodedVersionLength etc do not appear to be valid. We should throw a non-runtime exception.